### PR TITLE
Changed argument leafs to leaves, with deprecation warning in ontograph

### PR DESCRIPTION
**Pull request by francescalb**
# Description:
The plural of leaf was written leafs in all arguments etc. Changed it to leaves.

Added a deprecation warning on the tool ontograph, but apart from that did not update with deprecation warnings after discussing the need with @jesper-friis.

## Type of change:

- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist:

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

File: ontopy/ontodoc.py [hunk 1]
```diff
@@ -518,35 +518,35 @@ class DocPP:  # pylint: disable=too-many-instance-attributes
           - header_level: Header level.
           - terminated: Whether to branch should be terminated at all branch
             names in the final document.
-          - include_leafs: Whether to include leaf.
+          - include_leaves: Whether to include leaf.
```
Comment (francescalb) [inline ontopy/ontodoc.py:521]:
```suggestion
          - include_leaves: Whether to include leaves as end point of the branch.
```
```suggestion
          - include_leaves: Whether to include leaves as end points to the branch.
```

File: ontopy/ontodoc.py [hunk 2]
```diff
@@ -776,11 +776,11 @@ def _make_branchfig(  # pylint: disable=too-many-arguments,too-many-locals
         Args:
             name: name of branch root
             path: optional figure path name
-            include_leafs: whether to include leafs
-            strict_leafs: whether strictly exclude leafs descendants
+            include_leaves: whether to include leaves
+            strict_leaves: whether strictly exclude leaves descendants
```
Comment (francescalb) [inline ontopy/ontodoc.py:780]:
```suggestion
            strict_leaves: whether to strictly exclude leave descendants
```

File: ontopy/ontodoc.py [hunk 3]
```diff
@@ -776,11 +776,11 @@ def _make_branchfig(  # pylint: disable=too-many-arguments,too-many-locals
         Args:
             name: name of branch root
             path: optional figure path name
-            include_leafs: whether to include leafs
-            strict_leafs: whether strictly exclude leafs descendants
+            include_leaves: whether to include leaves
```
Comment (francescalb) [inline ontopy/ontodoc.py:779]:
```suggestion
            include_leaves: whether to include leaves as end points to the branch.
```

File: ontopy/ontodoc.py [hunk 4]
```diff
@@ -518,35 +518,36 @@ class DocPP:  # pylint: disable=too-many-instance-attributes
           - header_level: Header level.
           - terminated: Whether to branch should be terminated at all branch
             names in the final document.
-          - include_leafs: Whether to include leaf.
+          - include_leaves: Whether to include leaves as end points
```
Comment (CasperWA) [inline ontopy/ontodoc.py:521]: Excellent with some extra clarification!! ❤️

File: tools/ontograph [hunk 5]
```diff
@@ -113,14 +114,20 @@ def main(
...
-            "comma-separated string and/or with multiple --leafs options."
+            "Leaf nodes for plotting sub-graphs. May be provided as a "
+            "comma-separated string and/or with multiple --leaves options."
         ),
     )
+    parser.add_argument(
+        "--leafs",
+        action="append",
+        default=[],
+        help=("This argument is deprecated, plaese use `--leaves` instead"),
```
Comment (CasperWA) [inline tools/ontograph:129]:
```suggestion
        help=("This argument is deprecated, please use `--leaves` instead"),
```

File: tools/ontograph [hunk 6]
```diff
@@ -209,6 +216,13 @@ def main(
     )
     args = parser.parse_args(args=argv)
 
+    if args.leafs:
+        warnings.warn(
+            "The `--leafs` option is deprecated and will soon be removed."
```
Comment (CasperWA) [inline tools/ontograph:221]:
```suggestion
            "The `--leafs` option is deprecated and will soon be removed. "
```

File: tools/ontograph [hunk 7]
```diff
@@ -209,6 +216,13 @@ def main(
...
+            "Please use `--leaves` instead."
```
Comment (CasperWA) [inline tools/ontograph:222]:
```suggestion
            "Please use `--leaves` instead.",
            DeprecationWarning,
```
Not sure this is correct and also whether `DeprecationWarning` is imported by default (like, e.g., `Exception`). But in this way the warning will be a "proper" deprecation warning and not a `UserWarning`, which is the default warning class.

**Review by CasperWA — CHANGES_REQUESTED** [review]
Excellent! :) You probably need to test the `DeprecationWarning` thing locally, once implemented (see comment below).

**Review by CasperWA — APPROVED** [review]
I guess the deprecation warning could be tested (and tested that it still works with `leafs`) but I don't consider it a must, to be honest.

**Pull request merged**